main.tf

```diff
@@ -1,34 +1,43 @@
-# # TODO: Replace this dummy resource azurerm_resource_group.TODO with your module resource
-# resource "azurerm_resource_group" "this" {
-#   location = var.location
-#   name     = var.name # calling code must supply the name
-#   tags     = var.tags
-# }
 
-# # required AVM resources interfaces
-# resource "azurerm_management_lock" "this" {
-#   count = var.lock != null ? 1 : 0
+# resource "azurerm_api_management" "this" {
+#   name                = var.name
+#   location            = var.location
+#   resource_group_name = var.resource_group_name
+#   publisher_name      = var.publisher_name
+#   publisher_email     = var.publisher_email
+#   sku_name           = var.sku_name
+#   tags               = var.tags
+#   # This implementation uses a dynamic block with for_each to conditionally create the virtual_network_configuration block only when virtual_network_type is either "Internal" or "External".
+#   # If the type is "None", the block won't be included in the resource.
+#   dynamic "virtual_network_configuration" {
+#     for_each = contains(["Internal", "External"], var.virtual_network_type) ? [1] : []
+#     content {
+#       subnet_id = var.virtual_network_subnet_id
+#     }
+#   }
+#   virtual_network_type = var.virtual_network_type
+#   # min_api_version = "2024-06-01-preview"
 
```

Comment:
are we keeping this commented out apim resource for any particular reason?

Reply:
no, will delete it